bin/cylc-cat-log

```diff
@@ -29,227 +29,276 @@ if remrun().execute():
 from cylc.suite_host import is_remote_host
-from cylc.suite_logging import suite_log
-from cylc.suite_output import suite_output
 from cylc.cfgspec.globalcfg import GLOBAL_CFG
 from cylc.task_id import TaskID
 
 
-def main():
+NAME_DEFAULT = "log"
+NAME_ERR = "err"
+NAME_OUT = "out"
+NAME_JOB_ACTIVITY_LOG = "job-activity.log"
+NAME_JOB_EDIT_DIFF = "job-edit.diff"
+NAME_JOB_STATUS = "job.status"
+JOB_LOG_DEST_MAP = {
+    NAME_DEFAULT: "job", NAME_ERR: "job.err", NAME_OUT: "job.out"}
+JOB_LOG_LOCAL_ALWAYS = (NAME_JOB_EDIT_DIFF, NAME_JOB_ACTIVITY_LOG)
+LIST_MODE_LOCAL = "local"
+LIST_MODE_REMOTE = "remote"
+
 
+def get_option_parser():
+    """Set up the CLI option parser."""
     parser = cop(
         __doc__, argdoc=[("REG", "Suite name"), ("[TASK-ID]", """Task ID""")])
 
     parser.add_option(
         "-l", "--location",
-        help="Print the location of the file, and set exit status according"
-        "to whether or not the file exists (0 means it exists).",
-        action="store_true", default=False, dest="location")
+        help=("Print location of the log file, exit 0 if it exists," +
+              " exit 1 otherwise"),
+        action="store_true", default=False, dest="location_mode")
 
     parser.add_option(
         "-o", "--stdout",
-        help="Suite or task stdout log.",
-        action="store_true", default=False, dest="stdout")
+        help="Suite log: out, task job log: job.out",
+        action="store_const", const=NAME_OUT, dest="filename")
 
     parser.add_option(
         "-e", "--stderr",
-        help="Suite or task stderr log.",
-        action="store_true", default=False, dest="stderr")
+        help="Suite log: err, task job log: job.err",
+        action="store_const", const=NAME_ERR, dest="filename")
 
     parser.add_option(
         "-r", "--rotation",
-        help="Suite logs log rotation number (default 0)",
-        metavar="INT", action="store", default="0", dest="rotation")
+        help="Suite logs log rotation number", metavar="INT",
+        action="store", dest="rotation_num")
 
     parser.add_option(
         "-a", "--activity",
-        help="Task job activity log.",
-        action="store_true", default=False, dest="job_activity")
+        help="Task job log only: Short for --filename=job-activity.log",
+        action="store_const", const=NAME_JOB_ACTIVITY_LOG, dest="filename")
 
     parser.add_option(
         "-d", "--diff",
-        help="Task job script diff (present after an edit-run).",
-        action="store_true", default=False, dest="job_diff")
+        help=("Task job log only: Short for --filename=job-edit.diff" +
+              " (file present after an edit-run)."),
+        action="store_const", const=NAME_JOB_EDIT_DIFF, dest="filename")
 
     parser.add_option(
         "-u", "--status",
-        help="Task job status file.",
-        action="store_true", default=False, dest="job_status")
-
-    parser.add_option(
-        "-c", "--custom",
```

Comment:
Looks like we've lost the -c functionality here? is that ok e.g. can we now no longer view job.stats files with this cmd?